### PR TITLE
Create Service Class to Return UnMarshalled Object

### PR DESCRIPTION
**Pull request by lbrown-va**
Created hand-of-queen-elizabeth project service class named QueenElizabethService to encapsulate the eeSummary SOAP services that return appropriate objects of generated type.

### CODE REVIEW

File: hand-of-queen-elizabeth/src/main/java/gov/va/api/health/queenelizabeth/ee/QueenElizabethService.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  }
+
+  /**
+   * Request getEESummary from the getEESummary SOAP Service repeatedly for a list of ICN.
+   *
+   * @param icnList List of ICN.
+   * @return Linked List of GetEESummaryResponse.
+   */
+  public List<GetEESummaryResponse> getEeSummary(final List<String> icnList) {
+    // TODO: Does this really need to be a LinkedList?  LinkedList was specified by carma so I left
```
Comment (beaugrantham) [inline hand-of-queen-elizabeth/src/main/java/gov/va/api/health/queenelizabeth/ee/QueenElizabethService.java]: I don't see anything here that would behave differently than an ordinary `ArrayList`.

**Comment by beaugrantham-va** [comment]
> 
> 
> Looks good.

Other than the pom conflicts of course.

**Comment by esubocz-va** [comment]
Have you checked to see how these changes affect other projects that it uses outside of urgent-care? ie. community-care, provider-directory(maybe?), carma-experience?

**Comment by jblefkove-va** [comment]
I'm concerned about backwards-compatibility. To ease migration and avoid breaking the clients that use the `Eligibilities` interface, can we add that back with an implementation that delegates to `QueenElizabethService`?

File: hand-of-queen-elizabeth/src/test/resources/application.properties [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+
+#----------------------------------------------------------------------------
+# Unit Test Settings Only
+# Settings for the mock server instantiated for the unit test.
+#
+server.port = 9090
+security.require-ssl=true
+
+spring.datasource.url=jdbc:h2:mem:TEST;MVCC=true;DB_CLOSE_DELAY=-1;MODE=Oracle
```
Comment (alecbert-va) [inline hand-of-queen-elizabeth/src/test/resources/application.properties]: what requires the datasource, the spring app for serving mock data?

Comment (lbrown-va) [reply hand-of-queen-elizabeth/src/test/resources/application.properties]: I believe this was a silly cut/paste when I added the mock server.  I only server.port is necessary.

**Comment by alecbert-va** [comment]
> I'm concerned about backwards-compatibility. To ease migration and avoid breaking the clients that use the `Eligibilities` interface, can we add that back with an implementation that delegates to `QueenElizabethService`?

@jblefkove-va I thought the deal was that there'd be a major version bump and apps could adopt the new client as they had time. If the major version bump won't be enough for Jenkins to not try and auto update dependencies let me know. The purpose for Eligibilities should be replaced by hand of QE returning a getEESummary Java object directly. 

There's a branch on CCE that uses this version of the client and shows it working at least with mock-ee, I'm attempting to set up my GFE for dev use again so I can connect to the QA endpoints directly for that last bit of assurance.

**Pull request merged**